src/main/java/com/purbon/kafka/topology/api/ksql/KsqlApiClient.java

```diff
@@ -11,38 +12,48 @@
 import io.confluent.ksql.api.client.ClientOptions;
 import io.confluent.ksql.api.client.StreamInfo;
 import io.confluent.ksql.api.client.TableInfo;
+
+import javax.annotation.Nullable;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
+import java.net.URL;
+import java.util.*;
 import java.util.concurrent.ExecutionException;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public class KsqlApiClient implements ArtefactClient {
 
-  private String server;
-  private Integer port;
-  private Client client;
-
-  public static String QUERY_TYPE = "query";
-  public static String STREAM_TYPE = "stream";
-  public static String TABLE_TYPE = "table";
-
-  public KsqlApiClient(String server, Integer port) {
+  private final URL server;
+  private final Client client;
+
+  public static final String HTTPS = "https";
+  public static final String QUERY_TYPE = "query";
+  public static final String STREAM_TYPE = "stream";
+  public static final String TABLE_TYPE = "table";
+
+  public KsqlApiClient(URL server, @Nullable Configuration.BasicAuth basicAuth) {
```

Comment:
might be use Optional?

Reply:
Optional is primarily intended as a return type, not as an argument.
see https://docs.oracle.com/javase/10/docs/api/java/util/Optional.html